tensorboard/examples/plugins/example_basic/tensorboard_plugin_example/plugin.py

```diff
@@ -74,12 +74,21 @@ def _serve_tags(self, request):
 
   @wrappers.Request.application
   def _serve_greetings(self, request):
-    run = request.args["run"]
-    tag = request.args["tag"]
-    data = [
-        np.asscalar(tensor_util.make_ndarray(event.tensor_proto))
-            .decode("utf-8")
-        for event in self._multiplexer.Tensors(run, tag)
-    ]
+    run = request.args.get("run")
+    tag = request.args.get("tag")
+    if run is None or tag is None:
+      return werkzeug.Response(
+          "Must specify run and tag",
+          content_type="text/plain",
+          code=400,
+      )
+    try:
+      data = [
+          np.asscalar(tensor_util.make_ndarray(event.tensor_proto))
+              .decode("utf-8")
+          for event in self._multiplexer.Tensors(run, tag)
+      ]
+    except KeyError:
+      raise werkzeug.exceptions.BadRequest("Invalid run or tag")
```

Comment:
just wondering, HTTP 400 seems to be the BAD_REQUEST. Why do we opt out of using `werkzeug.exceptions.BadRequest` in line 80?

Reply:
…I can’t remember why I might have done that. Will change them both to
use the exception formulation. Thanks for the catch.

(Prior to this patch, lines 77 and 78 actually were already responding
with 400 Bad Request on failure (`request.args` is a magic container
whose key error type is `exceptions.BadRequest`), but I thought that
they should get a nicer message. But they should use the same form.)